### PR TITLE
Airport Challenge

### PR DESCRIPTION
**Pull request by ggwc82**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -56,24 +67,3 @@ Your code should defend against [edge cases](http://programmers.stackexchange.co
 
 For overriding random weather behaviour, please read the documentation to learn how to use test doubles: https://www.relishapp.com/rspec/rspec-mocks/docs . There’s an example of using a test double to test a die that’s relevant to testing random weather in the test.
```
Comment (RPiper93) [inline README.md]: You can add instructions on how to use your code.

File: spec/airport_spec.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  let(:plane) { double :plane}
+
+
+  describe "#capacity" do
+    it "tests default capacity set to #{Airport::DEFAULT_CAPACITY}" do 
+      expect(subject.capacity).to eq(Airport::DEFAULT_CAPACITY)
+    end
+
+    it "tests that capacity can be changed to 30" do
+      subject.capacity = 30
```
Comment (RPiper93) [inline spec/airport_spec.rb]: You're using the implicit subject. You can declare the subject at the top and use a named subject throughout the test.

File: lib/plane.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  def takeoff(airport, weather)
+    fail "Plane already airborne" unless landed
+    fail "Plane not at this airport" unless airport == @airport
+    fail "Can't take off - weather is stormy" if weather.condition == "stormy"
+    @landed = false
+    airport.planes.delete(self)
+    self
+  end
+
+end
```
Comment (RPiper93) [inline lib/plane.rb]: Could you put weather.condition == "stormy" into a private method? How about airport == @airport?

File: lib/plane.rb [hunk 4]
```diff
@@ -0,0 +1,28 @@
+class Plane
+
+  attr_reader :landed, :airport
+
+  def initialize
+    @airport
+  end
```
Comment (RPiper93) [inline lib/plane.rb]: Do you think this code is necessary? What value would @ airport have if you didn't initialize it?

**Comment by mechacoach** [comment]
You had your airport_challenge reviewed by **Rebecca Piper**

### The good points are:

* Overview of project in the README
* The solution avoids vacuous tests
* There are a minimal number of untested paths in the code
* Unit tests all have single expectations
* Random behaviour is stubbed where appropriate
* The code follows Ruby naming conventions, Readability is supported by consistent naming
* Guard clauses are used where appropriate, Booleans are returned implicitly where appropriate, removeallcommented-outcode: There is no commented-out code in the submission
* Classes have single responsibilities, Each methods has a single responsibility
* Code avoids strings where it should use symbols
* Command/Query naming is not an issue in the code

### You should consider the following improvements:

* Installation Instructions should be in the README
* Use named subject with `described_class`
* Use `context` and `describe` blocks to create test scopes
* Use `before` blocks to set up objects rather than repeat code
* Get rid of redundant `respond_to` expectations
* Remove all commented-out code
* Do not expose internal implementation
* Avoid magic numbers (e.g. on capacity)
* Improper use of attr_*
* Avoid redundant lines of code

### Additional comments:
Step 3: Uses let appropriately but didn't declare subject. Used 'describe' blocks but not 'context'.
Step 4: Only had about one line of redundant code.
, avoidmagicnumberse.g.oncapacity: Code avoids magic numbers

see https://github.com/makersacademy/airport_challenge/blob/master/docs/review.md for more details

**Comment by mechacoach** [comment]
You had your airport_challenge reviewed by **Rebecca Piper**

### The good points are:

* Overview of project in the README
* The solution avoids vacuous tests
* There are a minimal number of untested paths in the code
* Unit tests all have single expectations
* Random behaviour is stubbed where appropriate
* The code follows Ruby naming conventions, Readability is supported by consistent naming
* Guard clauses are used where appropriate, Booleans are returned implicitly where appropriate, removeallcommented-outcode: There is no commented-out code in the submission
* Classes have single responsibilities, Each methods has a single responsibility
* Code avoids strings where it should use symbols
* Command/Query naming is not an issue in the code

### You should consider the following improvements:

* Installation Instructions should be in the README
* Use named subject with `described_class`
* Use `context` and `describe` blocks to create test scopes
* Use `before` blocks to set up objects rather than repeat code
* Get rid of redundant `respond_to` expectations
* Remove all commented-out code
* Do not expose internal implementation
* Avoid magic numbers (e.g. on capacity)
* Improper use of attr_*
* Avoid redundant lines of code

### Additional comments:
Step 3: Uses let appropriately but didn't declare subject. Used 'describe' blocks but not 'context'.
Step 4: Only had about one line of redundant code.
, avoidmagicnumberse.g.oncapacity: Code avoids magic numbers

see https://github.com/makersacademy/airport_challenge/blob/master/docs/review.md for more details

**Pull request closed without merging**